app/models/repository.rb

```diff
@@ -120,38 +121,49 @@ def self.add_repo(event, namespace, repo, tag)
       repository = Repository.create(namespace: namespace, name: repo)
     elsif repository.tags.exists?(name: tag)
       # Update digest and status if the given tag already exists.
-      id, digest = Repository.id_and_digest_from_event(event, repository.full_name)
+      data = Repository.data_from_event(event, repository.full_name)
       tag = repository.tags.find_by(name: tag)
-      tag.update_columns(image_id:   id,
-                         digest:     digest,
+      tag.update_columns(image_id:   data.id,
+                         digest:     data.digest,
+                         size:       data.size,
                          scanned:    Tag.statuses[:scan_none],
                          updated_at: Time.current)
       repository.create_activity(:push, owner: actor, recipient: tag)
       return
     end
 
     # And store the tag and its activity.
-    id, digest = Repository.id_and_digest_from_event(event, repository.full_name)
-    tag = repository.tags.create(name: tag, author: actor, digest: digest, image_id: id)
+    data = Repository.data_from_event(event, repository.full_name)
+    tag = repository.tags.create(name:     tag,
+                                 author:   actor,
+                                 digest:   data.digest,
+                                 image_id: data.id,
+                                 size:     data.size)
     repository.create_activity(:push, owner: actor, recipient: tag)
     repository
   end
+  # rubocop:enable Metrics/MethodLength
 
-  # Fetch the image ID and the manifest digest from the given event.
-  def self.id_and_digest_from_event(event, repo)
+  # Fetch the image ID and the manifest digest from the given event and returns
+  # an OpenStruct object containing the following attributes:
+  #
+  #   - id:     The image ID (without the "sha256:" prefix)
+  #   - digest: The manifest digest
+  #   - size:   The tag size
+  def self.data_from_event(event, repo)
     digest = event.try(:[], "target").try(:[], "digest")
-    id = ""
+    manifest = OpenStruct.new(id: "", digest: digest, size: nil)
 
     if digest.present?
       begin
-        id, = Registry.get.client.manifest(repo, digest)
+        manifest = Registry.get.client.manifest(repo, digest)
       rescue ::Portus::RequestError, ::Portus::Errors::NotFoundError,
              ::Portus::RegistryClient::ManifestError => e
         logger.warn "Could not fetch manifest for '#{repo}' with digest '#{digest}': " + e.to_s
       end
     end
 
-    [id, digest]
+    OpenStruct.new(id: manifest.id, digest: digest, size: manifest.size)
```

Comment:
Couldn't you just reuse the `manifest ` object? (while updating what it's needed)

Reply:
I thought about that but this would be one liner and not passing the manifest along. I assumed that the manifest wasn't that important for whoever was calling this. Otherwise it's no big deal to include it.

Reply:
I mean, I'd assign whatever is new, and then return the object. My point is to reuse the object that we create right at the beginning: we want to avoid creating new objects if we can. I remember that the last issue we got about performance there was something about memory going up. We can avoid these kinds of situations by reusing the object and then return it.

Reply:
Ok, fair enough...